internal/repository/memory_test.go

```diff
@@ -33,6 +33,26 @@ func TestInMemoryRepositoryRegistration(t *testing.T) {
 	assert.Equal(t, m2, repo)
 }
 
+func TestInMemoryRepositoryParsingWithEmptyList(t *testing.T) {
+	m := NewInMemoryRepository("000")
+	repository.Register("dht", m)
+	m.Data["/foo"] = []byte{1, 2, 3}
+	m.Data["/foo/bar"] = []byte{1, 2, 3}
+	m.Data["/foo/baz/"] = []byte{1, 2, 3}
+	m.Data["/foo/baz/quux"] = []byte{1, 2, 3}
+
+	foo, err := storage.ParseURI("dht:?00000")
+	assert.Nil(t, err)
+
+	canList, err := storage.CanList(foo)
+	assert.NotNil(t, err)
```

Comment:
Is this right? Asking if it can list returning an error...
Then the next call to list on something that cannot is checking for no error. Seems backwards somehow.

Reply:
Or i do not need to call this one?
Because root case is  in `List` of InMemoryRepository
prefix from path is return as empty string then when it called`prefix[len(prefix)-1]` It will be out of bound.

Reply:
Or can i delete the test case ?

Reply:
I think I understand your case here. The data is just rather hard to understand as you have a custom scheme and a custom structure of data - perhaps something more traditional that performs the same test?